source/ota.c

```diff
@@ -2772,41 +2771,18 @@ OtaState_t OTA_AgentShutdown( uint32_t ticksToWait )
 /*
  * Return the current state of the OTA agent.
  */
-OtaState_t OTA_GetAgentState( void )
+OtaState_t OTA_GetState( void )
 {
     return otaAgent.state;
 }
 
 /*
- * Return the number of packets dropped.
+ * Return the details of the packets received.
  */
-uint32_t OTA_GetPacketsDropped( void )
+OtaErr_t OTA_GetStatistics( OtaAgentStatistics_t * pStatistics )
 {
-    return otaAgent.statistics.otaPacketsDropped;
-}
-
-/*
- * Return the number of packets queued.
- */
-uint32_t OTA_GetPacketsQueued( void )
-{
-    return otaAgent.statistics.otaPacketsQueued;
-}
-
-/*
- * Return the number of packets processed.
- */
-uint32_t OTA_GetPacketsProcessed( void )
-{
-    return otaAgent.statistics.otaPacketsProcessed;
-}
-
-/*
- * Return the number of packets received.
- */
-uint32_t OTA_GetPacketsReceived( void )
-{
-    return otaAgent.statistics.otaPacketsReceived;
+    *pStatistics = otaAgent.statistics;
```

Comment:
Since this is a public API, you need to validate `pStatistics` before dereferencing it, e.g.:

```
OtaErr_t returnStatus = OTA_ERR_INVALID_PARAMS;
if( pStatistics != NULL )
{
    *pStatistics = otaAgent.statistics;
    returnStatus = OTA_ERR_NONE;
}
return returnStatus;

Reply:
Thanks, updated:
https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/9b8ba39796aab5af8d52b3423c4a497ae0fbbe4a/source/ota.c#L2782